codalab/worker/worker.py

```diff
@@ -759,6 +772,23 @@ def bundle_service_reply(self, socket_id, err, message, data):
         else:
             self.bundle_service.reply(self.id, socket_id, message)
 
+    def save_time_stats(self, uuid, from_stage):
+        self.end_stage(uuid, from_stage.stage)
+        self.start_stage(uuid, self.runs[uuid].stage)
+
+    def start_stage(self, uuid, stage):
+        self.runs[uuid].bundle_profile_stats[stage]['start'] = time.time()
+
+    def end_stage(self, uuid, stage):
+        self.runs[uuid].bundle_profile_stats[stage]['end'] = time.time()
+        self.runs[uuid].bundle_profile_stats[stage]['elapsed'] = (
+            self.runs[uuid].bundle_profile_stats[stage]['end']
+            - self.runs[uuid].bundle_profile_stats[stage]['start']
+        )
+
+    def initstats(self):
+        return {'start': 0, 'end': 0, 'elapsed': -1}
```

Comment:
Why -1 and not 0?

Reply:
i wanted to have it marked as an "impossible" value, elapsed can technically be 0

Reply:
Maybe `None`?